SourceCode/Power/ftt_p_shares.py

```diff
@@ -145,59 +145,75 @@ def shares(dt, t, T_Scal, mewdt, mwdlt, mews_dt, metc_dt, mtcd_dt,
         # Add in exogenous sales figures. These are blended with
         # endogenous result! Note that it's different from the
         # ExogSales specification!
-        Utot = np.sum(mewk_dt[:, :, 0], axis=1)
 
         dSk = np.zeros((t2ti))
         dUk = np.zeros((t2ti))
         dUkTK = np.zeros((t2ti))
         dUkREG = np.zeros((t2ti))
 
 
+        endo_shares = mews_dt[r, :, 0] + np.sum(dSik, axis=1)
+
+        # Copy over load factors that do not change
+        # Only applies to baseload and variable technologies
+        mewl[r, :, 0] = mewl_dt[r, :, 0].copy()
+        # new_capacity_idx = np.logical_and(mews_lag[r, :, 0]==0, mews[r, :, 0] > 0)
+        for tech_idx in range(t2ti):
+            if np.logical_and(mews_lag[r, tech_idx, 0]==0, endo_shares[tech_idx] > 0):
+                    mewl[r, tech_idx, 0] = mwlo[r, tech_idx, 0]
+
+        endo_gen = endo_shares * (mewdt[r]*1000/3.6) * mewl[r, :, 0] / np.sum(endo_shares * mewl[r, :, 0])
+
+        endo_capacity = endo_gen / mewl[r, :, 0] / 8766
+
+        Utot = np.sum(endo_capacity)
+        
+
+
+
         # PV: Added a term to check that exogenous capacity is smaller than regulated capacity.
         # Regulations have priority over exogenous capacity
         reg_vs_exog = ((mwka[r, :, 0]) > mewr[r, :, 0]) & (mewr[r, :, 0] >= 0.0)
         mwka[r, :, 0] = np.where(reg_vs_exog, -1.0, mwka[r, :, 0])
         MWKA_scalar = 1.0
 
-        dUkTK = (mwka[r, :, 0] - mewk_lag[r, :, 0])/no_it
+
+
+        # Correct for regulations using difference between endogenous capacity and capacity from last time step with endo shares
+            
+        dUkREG = -(endo_capacity - endo_shares*np.sum(mewk_dt[r, :, 0]))* isReg[r, :] 
+
+        #calculate capcity additions or subtractions after regulations, to prevent subtractions being too large and causing negatve shares.
```

Comment:
```suggestion
        # calculate capacity additions or subtractions after regulations, to prevent subtractions being too large and causing negatve shares.
```